xlb/operator/boundary_masker/indices_boundary_masker.py

```diff
@@ -47,12 +47,18 @@ def jax_implementation(self, bclist, bc_mask, missing_mask, start_index=None):
         dim = missing_mask.ndim - 1
         nDevices = jax.device_count()
         pad_x, pad_y, pad_z = nDevices, 1, 1
+        # TODO MEHDI: There is sometimes a halting problem here when padding is used in a multi-GPU setting since we're not jitting this function.
+        # For now, we compute the bmap on GPU zero.
         if dim == 2:
+            bmap = jnp.zeros((pad_x * 2 + bc_mask[0].shape[0], pad_y * 2 + bc_mask[0].shape[1]), dtype=jnp.uint8)
```

Comment:
Are these changes related to the issue you have just opened? if yes, is this a fix? if yes then please link that issue to the PR.

Reply:
This is not a fix, but needed to stop halting otherwise nothing works on JAX if you have more than 1 GPU.